plugins/modules/keycloak_realm.py

```diff
@@ -767,9 +766,6 @@ def main():
         # Process a creation
         result['changed'] = True
 
-        if 'id' not in desired_realm:
-            module.fail_json(msg='id needs to be specified when creating a new realm')
```

Comment:
Right now you can run the module with neither `id` nor `realm` specified. That doesn't sound useful.

(Also it doesn't seem like the module actually works well if `realm` isn't specified?)

Reply:
I'm fine with making `realm` required. Since the module tries to determine whether a realm exists by looking for it, I don't think it really works (or ever worked) without `realm` anyway.